File: Number Operation/prg.scala
object LogicalBinaryOps extends App {
    val a = scala.io.StdIn.readInt()
    val b = scala.io.StdIn.readInt()

    println(s"AND: ${a & b}")
    println(s"OR: ${a | b}")
    println(s"XOR: ${a ^ b}")
    println(s"NOT A: ${~a}")
    println(s"Left Shift: ${a << 1}")
    println(s"Right Shift: ${a >> 1}")

    val x = scala.io.StdIn.readBoolean()
    val y = scala.io.StdIn.readBoolean()
    println(s"Logical AND: ${x && y}")
    println(s"Logical OR: ${x || y}")
    println(s"Logical NOT X: ${!x}")
}

File: Number Operation/prg.cpp
#include <iostream>
using namespace std;

int main() {
    int a, b;
    cin >> a >> b;

    cout << "AND: " << (a & b) << endl;
    cout << "OR: " << (a | b) << endl;
    cout << "XOR: " << (a ^ b) << endl;
    cout << "NOT A: " << (~a) << endl;
    cout << "Left Shift: " << (a << 1) << endl;
    cout << "Right Shift: " << (a >> 1) << endl;

    bool x, y;
    cin >> x >> y;
    cout << "Logical AND: " << (x && y) << endl;
    cout << "Logical OR: " << (x || y) << endl;
    cout << "Logical NOT X: " << (!x) << endl;

    return 0;
}
